src/dispatch/experimental/lambda_handler.py

```diff
@@ -0,0 +1,126 @@
+"""Integration of Dispatch programmable endpoints for FastAPI.
+
+Example:
+
+    from dispatch.experimental.lambda_handler import Dispatch
+
+    dispatch = Dispatch(api_key="test-key")
+
+    @dispatch.function
+    def my_function():
+        return "Hello World!"
+
+    @dispatch.entrypoint
+    def entrypoint():
+        my_function()
+
+    def handler(event, context):
+        dispatch.handle(event, context)
+    """
+
+import base64
+import json
+import logging
+from typing import Optional
+
+from dispatch.function import Registry
+from dispatch.proto import Input
+from dispatch.sdk.v1 import function_pb2 as function_pb
+from dispatch.status import Status
+
+logger = logging.getLogger(__name__)
+
+
+class Dispatch(Registry):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        api_url: Optional[str] = None,
+    ):
+        """Initializes a Dispatch Lambda handler.
+
+        Args:
+            api_key: Dispatch API key to use for authentication. Uses the value of
+                the DISPATCH_API_KEY environment variable by default.
+
+            api_url: The URL of the Dispatch API to use. Uses the value of the
+                DISPATCH_API_URL environment variable if set, otherwise
+                defaults to the public Dispatch API (DEFAULT_API_URL).
+
+        """
+
+        super().__init__(endpoint="not configured", api_key=api_key, api_url=api_url)
+
+    def handle(self, event, context):
+        # The ARN is not none until the first invocation of the Lambda function.
+        # We override the endpoint of all registered functions before any execution.
+        self.endpoint = context.invoked_function_arn
+        self.override_endpoint(self.endpoint)
+
+        if not event:
+            raise ValueError("event is required")
+
+        try:
+            raw = base64.b64decode(event)
+        except Exception as e:
+            raise ValueError(f"event is not base64 encoded: {e}")
```

Comment:
We could use exception chaining here instead of just the original message:
```suggestion
            raise ValueError(f"event is not base64 encoded") from e
```

Reply:
TIL `from e` thanks